legacy_packages/sdk/src/evm/common/any-evm-utils/getCreate2FactoryAddress.ts

```diff
@@ -16,23 +17,75 @@ import { getCreate2FactoryDeploymentInfo } from "./getCreate2FactoryDeploymentIn
 export async function getCreate2FactoryAddress(
   provider: providers.Provider,
 ): Promise<string> {
-  const commonFactoryExists = await isContractDeployed(
-    COMMON_FACTORY,
-    provider,
+  const deploymentInfo = await computeCreate2FactoryTransaction(provider);
+  return deploymentInfo.deployment;
+}
+
+/**
+ * @internal
+ */
+export async function computeCreate2FactoryTransaction(
+  provider: providers.Provider,
+): Promise<KeylessDeploymentInfo> {
+  const networkId = (await provider.getNetwork()).chainId;
+
+  // special handling for chains with hardcoded gasPrice and gasLimit
+  if (CUSTOM_GAS_FOR_CHAIN[networkId]) {
+    const enforceEip155 = await isEIP155Enforced(provider);
+    const chainId = enforceEip155 ? networkId : 0;
+    const gasPrice = CUSTOM_GAS_FOR_CHAIN[networkId]?.gasPrice;
+    const gasLimit = CUSTOM_GAS_FOR_CHAIN[networkId]?.gasLimit;
+
+    const deploymentInfo = getCreate2FactoryDeploymentInfo(chainId, {
+      gasPrice,
+      gasLimit,
+    });
+
+    if (await isContractDeployed(deploymentInfo.deployment, provider)) {
+      deploymentInfo.transaction = "";
+    }
+
+    return deploymentInfo;
+  }
+
+  // default flow
+  const allBinsInfo = [
+    ...CUSTOM_GAS_BINS.map((b) =>
+      // to generate EIP-155 transaction
+      getCreate2FactoryDeploymentInfo(networkId, { gasPrice: b }),
+    ),
+    // to generate pre-EIP-155 transaction, hence chainId 0
+    ...CUSTOM_GAS_BINS.map((b) =>
+      getCreate2FactoryDeploymentInfo(0, { gasPrice: b }),
+    ),
+  ];
+
+  const allFactories = await Promise.all(
+    allBinsInfo.map((b) => isContractDeployed(b.deployment, provider)),
```

Comment:
can this be multicalled?

Reply:
Multicalled meaning?
Right now it's being sent as a batch of promises, parallely.

Reply:
yeah this will end up batched into 1 rpc request, not sure if we can `multicall` a `eth_getCode` call further?